gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/search/DistributedLuceneIndexer.java

```diff
@@ -20,17 +20,20 @@
 import io.gravitee.rest.api.service.common.ExecutionContext;
 import io.gravitee.rest.api.service.search.SearchEngineService;
 import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
 import org.springframework.stereotype.Service;
 
 /**
  * @author Antoine CORDIER (antoine.cordier at graviteesource.com)
  * @author GraviteeSource Team
  */
-@RequiredArgsConstructor
 @Service
 public class DistributedLuceneIndexer implements Indexer {
 
-    private final SearchEngineService searchEngineService;
+    @Lazy
+    @Autowired
+    private SearchEngineService searchEngineService;
```

Comment:
can you prefer to write the constructor manually and use the annotation at parameter level ?

Reply:
no 😛